### PR TITLE
Issue #236: Updates to handle stand-alone UNIQUE and UNIQUE KEY statements

### LINKED ISSUE #236: Problems with UNIQUE parsing
There appears to be a couple of problems with UNIQUE parsing.

1. https://github.com/xnuinside/simple-ddl-parser/blob/f605740c55fcbc3a28cbfc682308fc5a1e1a21fe/tests/test_simple_ddl_parser.py#L365 I believe that this test is incorrect.  The output should not have UNIQUE on the individual columns, but rather should have an entry in `constraints.uniques` with both columns listed in the columns.  See https://www.mysqltutorial.org/mysql-basics/mysql-unique-constraint/#:~:text=To%20define%20a%20UNIQUE%C2%A0%20constraint%20for%20two%20or%20more%20columns%2C%20you%20use%20the%20following%20syntax%3A
2. `UNIQUE KEY` statements currently cause the entire table to be ignored.  IE `UNIQUE KEY "fun_unique_key" ("fun_unique_column"),` In this case I think we'd like a single column to parse into a column UNIQUE statement, and multiple columns to parse into the `constraints.uniques` location.

### PR DESCRIPTION
**Pull request by cfhowes**
resolves #236

Updated parsing of `UNIQUE` and `UNIQUE KEY` clauses and statements.  Now when multiple columns are specified in a `UNIQUE` clause/statement it will be treated as a compound UNIQUE statement and put in the `constraints` section of the parse dict.  It also means that the individual columns will have `unique=False` since the individual column does not actually have a UNIQUE constraint.  If I have done this correctly it will match the SQL spec more closely then before.

### CODE REVIEW

File: simple_ddl_parser/dialects/sql.py [hunk 1]
```diff
@@ -324,6 +324,7 @@ def p_column(self, p: List) -> None:
             p[0] = {"index_stmt": True, "name": p[2]["type"], "columns": ""}
             return
         if p[1] and isinstance(p[1], dict) and p[1].get("index_stmt") is True:
+            # @TODO: if we are normalizing columns, we need to normalize them here too.
```
Comment (cfhowes) [inline simple_ddl_parser/dialects/sql.py:327]: sorry, my TODO's got added, but this is also on my list....

File: simple_ddl_parser/dialects/sql.py [hunk 2]
```diff
@@ -1606,9 +1638,22 @@ def p_expression_primary_key(self, p):
         p[0] = p[1]
 
     def p_uniq(self, p: List) -> None:
-        """uniq : UNIQUE LP pid RP"""
+        """uniq : UNIQUE LP pid RP
+        | UNIQUE KEY id LP pid RP
```
Comment (cfhowes) [inline simple_ddl_parser/dialects/sql.py:1642]: before `UNIQUE KEY` statements were causing the parser to (silently) error and the table to not be generated.

File: tests/dialects/test_mssql_specific.py [hunk 3]
```diff
@@ -226,7 +226,7 @@ def test_constraint_unique():
                         "references": None,
                         "size": None,
                         "type": "BIGINT",
-                        "unique": True,
+                        "unique": False,
```
Comment (cfhowes) [inline tests/dialects/test_mssql_specific.py:229]: All these changes were from the move to handle compound unique clauses/statements.

File: tests/test_simple_ddl_parser.py [hunk 4]
```diff
@@ -275,164 +275,6 @@ def test_parser_multiple_tables():
     assert DDLParser(ddl).run() == expected
 
 
-def test_unique_statement_in_columns():
```
Comment (cfhowes) [inline tests/test_simple_ddl_parser.py:278]: i moved the tests specific to `UNIQUE` statements into a new file.  If you dislike I can put them back here.

File: tests/test_unique.py [hunk 5]
```diff
@@ -0,0 +1,261 @@
+from simple_ddl_parser import DDLParser
+
+
+def test_unique_statement_in_columns():
```
Comment (cfhowes) [inline tests/test_unique.py:4]: 2/3 of these are copy-paste from test_simple_ddl_parser.py.  The last test here is the new one for `UNIQUE KEY` statements.

**Comment by xnuinside** [comment]
@cfhowes thanks for your PR, I can take a look on this weekends

**Comment by xnuinside** [comment]
@cfhowes thanks for amazing update

**Pull request merged**